src/stats/stats_printer.cpp

```diff
@@ -305,6 +320,58 @@ static const char *state2str(tcp_state state)
     }
 }
 
+// This function parses /proc/net/[udp|tcp] file getting requested value
+static int get_proc_net_value(const char *name, uint32_t node, enum proc_net_type type, void *value, size_t size)
+{
+	int ret = -1;
+	uint32_t rxq, txq, time_len, retr, inode;
+	int num, local_port, rem_port, sl, state, uid, timer_run, timeout;
+	char rem_addr[128], local_addr[128];
+	FILE *fd = NULL;
+	char line[8192];
+
+	if (!name || !value || !size) {
+		return -1;
+	}
+
+	fd = fopen(name, "r");
+	if (fd) {
+		do {
+			if (fgets(line, sizeof(line), fd)) {
+				num = sscanf(line,
+					 "%d: %64[0-9A-Fa-f]:%X %64[0-9A-Fa-f]:%X %X %X:%X %X:%X %X %d %d %u %*s\n",
+					 &sl, local_addr, &local_port,
+					 rem_addr, &rem_port, &state,
+					 &txq, &rxq, &timer_run, &time_len, &retr, &uid, &timeout, &inode);
+				if ((14 == num) && (node == inode)) {
```

Comment:
3. For every offloaded TCP socket you scan a non-offloaded TCP file which potentially may be large.